### ISSUE TITLE
Add an option for ColorVariable via SCSS functions

### ISSUE THREAD

**Issue by acconrad** [brigade/scss-lint#794]
The `ColorVariable` rule currently only recognizes direct SCSS variables as a way to handle code smells around colors. My org uses a function to achieve this because we have theming, so the colors will be referenced in a map rather than a single variable, an example:

``` scss
background-color: color('gray-40');
```

Unfortunately the rules still consider this to be a violation in spite of the fact that it is achieving the same result of storing the color away in some sort of variable.

Could we create a flag within `ColorVariable` that allows for functions as well as variables?

**Comment by artemartemov** [comment]
This would be a fantastic feature. We are also using color maps for our platform and lint will hit as an error.

We have a font map in our base: 
`$color-base-grey: #CDCDCD;

$palettes: (
	black: (
		base: #000000
	),

	blue: (
		xlight: #B6ECFF,
		base: #74DCFF,
		dark: #00AAE3,
		xdark: #249CC3
	),

	green: (
		base: #87DB88
	),

	grey: (
		base: $color-base-grey,
		xxxlight: lighten($color-base-grey,24%),
		xxlight: lighten($color-base-grey, 18%),
		xlight: lighten($color-base-grey, 15%),
		light: lighten($color-base-grey, 12%),
		medium: darken($color-base-grey, 18%),
		dark: darken($color-base-grey, 32%),
		xdark: darken($color-base-grey, 40%),
		xxdark: darken($color-base-grey, 55%),
		xxxdark: darken($color-base-grey, 63%),
	),

	pink: (
		base: #EC008C
	),

	red: (
		base: #F23A55
	),

	teal: (
		base: #008080
	),

	white: (
		base: #FFFFFF
	),

	yellow: (
		base: #F6F489
	)
);`

So we pass it in our block files such as: 

`$color-bg-lazy: palette( grey, light );`
or
`$color-bg-lazy: palette( black );`

It will also error those guys out. Would love to be able to ignore those instances. 

Also would be great to be able to whitelist certain properties such as box-shadows to allow rgba to be passed.

**Comment by srawlins** [comment]
@acconrad can you give a full example? I cannot recreate the error using `'gray-40'`, but I can using @artemartemov's (with an educated guess at the `palette` function):

```scss
$palettes: (
  black: #000,
  blue: #74dcff,
);

@function palette($color) {
  @return map-get($palettes, $color);
}

a {
  color: palette(blue); // ColorVariable error.
}
```

I just want to make sure that there isn't a warning with a string like `'gray-40'`...

**Comment by acconrad** [comment]
@srawlins I would get warnings with a string as well since `palette(blue)` could be also read as `palette('blue')`
